sklearn/ensemble/weight_boosting.py

```diff
@@ -1049,7 +1051,7 @@ def _boost(self, iboost, X, y, sample_weight):
         if not iboost == self.n_estimators - 1:
             sample_weight *= np.power(
                 beta,
-                (1. - error_vect) * self.learning_rate)
+                (1. - error_vect) * self.learning_rate).astype(np.float64)
```

Comment:
Here you should use 'as_float_array(..., copy=False)' to avoid upcasting float32 into float64.